homeassistant/components/coolmaster/climate.py

```diff
@@ -42,58 +34,64 @@
     """Set up the CoolMasterNet climate platform."""
     supported_modes = config_entry.data.get(CONF_SUPPORTED_MODES)
-    host = config_entry.data[CONF_HOST]
-    port = config_entry.data[CONF_PORT]
-    cool = CoolMasterNet(host, port=port)
-    devices = await hass.async_add_executor_job(cool.devices)
+    info = hass.data[DOMAIN][config_entry.entry_id][DATA_INFO]
 
-    all_devices = [_build_entity(device, supported_modes) for device in devices]
+    coordinator = hass.data[DOMAIN][config_entry.entry_id][DATA_COORDINATOR]
 
-    async_add_devices(all_devices, True)
+    all_devices = [
+        _build_entity(coordinator, unit_id, unit, supported_modes, info)
+        for (unit_id, unit) in coordinator.data.items()
+    ]
+
+    async_add_devices(all_devices)
 
 
 class CoolmasterClimate(ClimateEntity):
     """Representation of a coolmaster climate device."""
 
-    def __init__(self, device, supported_modes):
+    def __init__(self, coordinator, unit_id, unit, supported_modes, info):
         """Initialize the climate device."""
-        self._device = device
-        self._uid = device.uid
+        self._coordinator = coordinator
+        self._unit_id = unit_id
+        self._unit = unit
         self._hvac_modes = supported_modes
-        self._hvac_mode = None
-        self._target_temperature = None
-        self._current_temperature = None
-        self._current_fan_mode = None
-        self._current_operation = None
-        self._on = None
-        self._unit = None
-
-    def update(self):
-        """Pull state from CoolMasterNet."""
-        status = self._device.status
-        self._target_temperature = status["thermostat"]
-        self._current_temperature = status["temperature"]
-        self._current_fan_mode = status["fan_speed"]
-        self._on = status["is_on"]
-
-        device_mode = status["mode"]
-        if self._on:
-            self._hvac_mode = CM_TO_HA_STATE[device_mode]
-        else:
-            self._hvac_mode = HVAC_MODE_OFF
+        self._info = info
 
-        if status["unit"] == "celsius":
-            self._unit = TEMP_CELSIUS
-        else:
-            self._unit = TEMP_FAHRENHEIT
+    @property
+    def should_poll(self):
+        """No need to poll. Coordinator notifies entity of updates."""
+        return False
+
+    @property
+    def available(self):
+        """Return if entity is available."""
+        return self._coordinator.last_update_success
+
+    def _refersh_from_coordinator(self):
+        self._unit = self._coordinator.data[self._unit_id]
+        self.async_write_ha_state()
+
+    async def async_added_to_hass(self):
+        """When entity is added to hass."""
+        self.async_on_remove(
+            self._coordinator.async_add_listener(self._refersh_from_coordinator)
+        )
+
+    async def async_update(self):
+        """Update the entity.
+
+        Only used by the generic entity update service.
+        """
+        self._unit = await self._unit.refresh()
```

Comment:
```suggestion
        await self._coordinator.async_request_update()
```

Reply:
this isn't the same thing - requesting an update from the coordinator refreshes all units, instead of just this specific one.